normalize-own-data/quantile_normalize_own_data.Rmd

```diff
@@ -0,0 +1,207 @@
+---
+title: "Quantile normalizing your own data to make it more comparable to refine.bio data"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: J. Taroni for CCDL
+date: 2019
+---
+
+## Background
+
+Datasets that are downloaded from the web interface of 
+[refine.bio](http://www.refine.bio) are 
+[quantile normalized](https://en.wikipedia.org/wiki/Quantile_normalization) 
+because we allow users to aggregate multiple platforms (e.g., microarray 
+chips) and even multiple technologies.
+Quantile normalization (QN) ensures that the underlying distributions of samples
+are the same.
+
+If you are interested in comparing your own data to data you have downloaded
+from the refine.bio website, you may wish to quantile normalize it using the
+reference distribution we've used in refine.bio.
+
+You can read more about the reference distribution and how it is generated in 
+the [quantile normalization section](http://docs.refine.bio/en/latest/main_text.html#quantile-normalization) 
+of our [documentation](http://docs.refine.bio/en/latest/).
+
+In this example, we'll demonstrate how to obtain the reference distribution
+from the refine.bio API and apply it to your own data in R.
+
+### Dataset
+
+We used [`GSE39842`](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39842) 
+in another module of the GitHub repository of examples: [`AlexsLemonade/refinebio-examples/compare-processing`](https://github.com/AlexsLemonade/refinebio-examples/tree/master/compare-processing).
+
+The submitters used RMA to process their data. 
+We have already converted the gene identifiers to Ensembl gene ids and can
+obtain the TSV file with these data from GitHub, placing it in the `data` 
+directory we create below with `bash`.
+
+```{bash eval=FALSE}
+mkdir data
+wget --directory-prefix=data https://github.com/AlexsLemonade/refinebio-examples/raw/c97cad6a8342d9bb54f120dbd7c7f12b03e3f799/compare-processing/data/GSE39842_RMA_converted.tsv
+```
+
+## Obtaining a refine.bio QN target 
+
+We'll use the terms "QN target" and "reference distribution" interchangeably to
+refer to the vector of values we use to quantile normalize `GSE39842`.
+
+We can query the refine.bio API like so:
+
+```
+https://api.refine.bio/qn_targets/?organism=<SCIENTIFIC_NAME>
+```
+
+where we would replace `<SCIENTIFIC_NAME>` with the name of the species we are
+interested in.
+
+We'd query the API for the zebrafish QN target like so:
+
+```
+https://api.refine.bio/qn_targets/?organism=DANIO_RERIO
+```
+
+Here, we'll use the `jsonlite` R package to extract the information we need to
+obtain the TSV file with the zebrafish reference distribution values. 
+
+```{r}
+qn_target_info <- jsonlite::fromJSON("https://api.refine.bio/qn_targets/?organism=DANIO_RERIO")
```

Comment:
For `preprocessCore` package you set it up to install if they didn't have it installed. Maybe you can put the same here for `jsonlite`?